wallet/createtx.go

```diff
@@ -1547,15 +1528,11 @@ func (w *Wallet) purchaseTickets(ctx context.Context, op errors.Op,
 				return nil, err
 			}
 		} else {
-			addrVote = req.VotingAddress
-			if addrVote == nil && req.Mixing {
-				addrVote = w.ticketAddress
-			}
-			if addrVote == nil {
-				addrVote, _, err = stakeAddrFunc(op, req.SourceAccount, 1)
-				if err != nil {
-					return nil, err
-				}
+			// If the user hasn't specified a voting account, derive a voting
+			// address from the purchasing account.
+			addrVote, _, err = stakeAddrFunc(op, req.SourceAccount, 1)
```

Comment:
Do we need to call  `signingAddressAtIdx` here?

Reply:
I don't think so.  It's only called once above, and the address return result is immediately discarded.  It appears the only thing it might be doing is persisting the returned address to the database, but this should already be handled by stakeAddrFunc.

This function should probably just go away.

Reply:
Ah, no, stakeAddrFunc is being called with the internal branch, but signingAddressAtIndex hardcodes the external branch.  Seems this is needed for VSP usecases.